docs/notebook-demos/layers.ipynb

```diff
@@ -0,0 +1,1310 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "def add_one(cells, _):\n",
+    "    return cells + 1\n",
+    "\n",
+    "# Mapping with a single funciton\n",
+    "raster_layer.map_cells(add_one)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "def divide_two(cells, _):\n",
+    "    return (add_one(cells) / 2)\n",
+    "\n",
+    "# Chaning together two functions to be mapped\n",
+    "raster_layer.map_cells(divide_two)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Mapping Over Tiles\n",
+    "\n",
+    "Like `map_cells`, `map_tiles` maps a given function over all of the `Tile`s within the layer. It takes a function that expects a `Tile` and returns a `Tile`. Therefore, the input function's type signature would be this:\n",
+    "\n",
+    "```python\n",
+    "def input_function(tile: Tile) -> Tile\n",
+    "```\n",
+    "\n",
+    "**Note**: In order for this method to operate, the internal `RDD` first needs to be deserialized from Scala to Python and then serialized from Python back to Scala. Because of this, it is recommended to chain together all functions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "def minus_two(tile):\n",
+    "    return gps.Tile.from_numpy_array(tile.cells - 2, no_data_value=tile.no_data_value)\n",
+    "\n",
+    "raster_layer.map_tiles(minus_two)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Calculating the Histogram for the Layer\n",
+    "\n",
+    "It is possible to calculate the histogram of a layer either by using the `get_histogram` or the `get_class_histogram` method. Both of these methods produce a `Histogram`, however, the way the data is represented is within this histogram differs depending on the method used. `get_histogram` will produce a histogram whose values are `float`s. Whereas `get_class_histogram` returns a histogram whose values are `int`s.\n",
+    "\n",
+    "For more informaiton on the `Histogram` class, please see the `Histogram` [guide]."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Returns a Histogram whose underlying values are floats\n",
+    "tiled_raster_layer.get_histogram()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Returns a Histogram whose underlying values are ints\n",
+    "tiled_raster_layer.get_class_histogram()"
```

Comment:
That was a new feature that wasn't merged in at the time. It was added to master on Friday, so that code block should now work.